pkg/agent/billing/billing.go

```diff
@@ -157,20 +198,23 @@ func (s *metricsState) collect(logger *zap.Logger, store VMStoreForNode, metrics
 		if !vm.Status.Phase.IsAlive() || vm.Status.CPUs == nil {
 			continue
 		}
+		go collectMetricsForVM(ctx, logger, vm, metricsChan)
+		metricsToCollect += 1
+	}
+
+	for i := 0; i < metricsToCollect; i++ {
+		kv := <-metricsChan
```

Comment:
Directly receiving from a channel like this is scary. If we have a logic error, the program might deadlock.

Let's (a) introduce a timeout _within_ calls to `collect` (maybe e.g. half of the collection period?) and (b) select over whether that context/timeout expires here, while waiting to receive.

Reply:
Added a select based on the context's timeout.